py/lexer.c

```diff
@@ -693,13 +698,13 @@ mp_lexer_t *mp_lexer_new(qstr src_name, mp_reader_t reader) {
     lex->chr1 = reader.readbyte(reader.data);
     lex->chr2 = reader.readbyte(reader.data);
 
+    convert_crlf(lex);
+
     // if input stream is 0, 1 or 2 characters long and doesn't end in a newline, then insert a newline at the end
     if (lex->chr0 == MP_LEXER_EOF) {
         lex->chr0 = '\n';
     } else if (lex->chr1 == MP_LEXER_EOF) {
-        if (lex->chr0 == '\r') {
```

Comment:
Note that we can remove this condition because `convert_crlf()` ensures `chr0` is not `'\r'`.